### PR TITLE
Use a different lock for `netlinkSocketCache` in manager

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR makes use of a different lock for accesses to `netlinkSocketCache`. The end goal is to limit races between `GetNetlinkSocket` and other usages of Probe that go through the manager.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes and instructions on how this should be tested.

### CODE REVIEW

File: manager.go [hunk 1]
```diff
@@ -1823,11 +1825,8 @@ func (m *Manager) newCachedNetlinkSocket(nsHandle uint64, nsID uint32) (*Netlink
 
 // GetNetlinkSocket - Returns a netlink socket in the requested network namespace from cache or creates a new one.
 func (m *Manager) GetNetlinkSocket(nsHandle uint64, nsID uint32) (*NetlinkSocket, error) {
-	m.stateLock.Lock()
-	defer m.stateLock.Unlock()
-	if m.state < initialized {
-		return nil, ErrManagerNotInitialized
-	}
+	m.nscLock.Lock()
+	defer m.nscLock.Unlock()
```
Comment (brycekahle) [inline manager.go:1829]: Any reason to have these mutexes outside the function, rather than part of `getNetlinkSocket`?

Comment (paulcacheux) [reply manager.go:1829]: It's because other manager functions are calling the `getNetlinkSocket` function while already having the mutex locked

Comment (brycekahle) [reply manager.go:1829]: They have the `stateLock` mutext locked correct? Not the `nscLock` mutex?

Comment (paulcacheux) [reply manager.go:1829]: Oh you are absolutely right

Comment (paulcacheux) [reply manager.go:1825]: Should be good now

**Review by brycekahle — APPROVED** [review]

**Pull request closed without merging**